### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by nirtiac**
Cleans up the main code base for syntax, includes some small fixes as specified. Will need to be rebased after the fix_early_stopping is merged.

### CODE REVIEW

File: vulcanai/datasets/tabulardataset.py [hunk 1]
```diff
@@ -516,10 +513,12 @@ def convert_all_categorical_binary(self, list_only=False,
             return binary_cols
 
         for col, index in binary_cols:
+            if col in exception_columns:
+                continue
             di = {index[0]: 1.0, index[1]: 0.0}
             try:
                 self.df = self.df.replace({col: di})
-            except:
+            except (AssertionError, TypeError, ValueError):
```
Comment (rfratila) [inline vulcanai/datasets/tabulardataset.py]: perhaps this is too silent of a way to handle these errors?

File: vulcanai/models/basenetwork.py [hunk 2]
```diff
@@ -2,8 +2,6 @@
 """Defines the basenetwork class."""
 # Core imports
 import abc
-import torch
```
Comment (rfratila) [inline vulcanai/models/basenetwork.py]: you're removing these because they are imported on __init__ ?

File: vulcanai/models/basenetwork.py [hunk 3]
```diff
@@ -877,53 +878,43 @@ def run_test(self, data_loader, plot=False, save_path=None, pos_label=1,
...
-	difference in predicting various values for a given feature when predicting outcome. 
-
-	Parameters:
-	    network : BaseNetwork
-                Network descendant of BaseNetwork.
+                            p_output_path=None, **kwargs):
+        """Performs bootfold - estimation to identify whether training model
+        provides statistically significant
+        difference in predicting various values for a given feature when
+        predicting outcome.
```
Comment (rfratila) [inline vulcanai/models/basenetwork.py]: I think you may need an empty line between the description and the parameters?

File: vulcanai/models/basenetwork.py [hunk 4]
```diff
@@ -877,53 +878,43 @@ def run_test(self, data_loader, plot=False, save_path=None, pos_label=1,
...
             epochs : int
                 The number of epochs to train the network per fold.
-	    index_to_iter : string
-		Index of feature within data_loader who's values will be iterated to assess difference
-	    ls_feat_vals : list
-		List of values for feature provided in feat_to_iter
+            index_to_iter : string
+                Index of feature within data_loader who's values will be
+                iterated.
+            to assess difference
```
Comment (rfratila) [inline vulcanai/models/basenetwork.py]: this line needs to be indented?

File: vulcanai/models/basenetwork.py [hunk 5]
```diff
@@ -877,53 +878,43 @@ def run_test(self, data_loader, plot=False, save_path=None, pos_label=1,
     def bootfold_p_estimate(self, data_loader, n_samples, k, epochs, 
                             index_to_iter, ls_feat_vals, retain_graph=None,
                             valid_interv=4, plot=False, save_path=None,
-                            transform_outputs=False, transform_callable=None, p_output_path = None,
```
Comment (rfratila) [inline vulcanai/models/basenetwork.py]: is the transform_callable parameter deprecated?

File: vulcanai/models/layers.py [hunk 6]
```diff
@@ -120,9 +122,7 @@ def __init__(self, in_features, out_features,
         # Main layer
         self._kernel = nn.Linear(
                             in_features=self.in_features,
-                            out_features=self.out_features,
-                            bias=True
```
Comment (rfratila) [inline vulcanai/models/layers.py]: you're removing the `bias=True` because that's the default value right? so it's redundant? just want to make sure

File: vulcanai/models/layers.py [hunk 7]
```diff
@@ -301,6 +301,7 @@ def __init__(self, in_channels, out_channels, bias=False):
 
     def forward(self, input):
```
Comment (rfratila) [inline vulcanai/models/layers.py]: yeah can make sure that this class is hidden so it doesn't automatically get used because it's not close to being ready and it's a bit of a vestigial organ for this file? Thank you

File: vulcanai/models/metrics.py [hunk 8]
```diff
@@ -45,11 +45,12 @@ def get_score(targets, predictions, metrics='accuracy', average=None,
...
             average: string,
-                [None, ‘binary’ (default), ‘micro’, ‘macro’, ‘samples’, ‘weighted’]
+                [None, ‘binary’ (default), ‘micro’, ‘macro’, ‘samples’,
+                ‘weighted’]
                 This parameter is required for multiclass/multilabel targets.
                 If None, the scores for each class are returned.
-                Otherwise, this determines the type of averaging performed on the
-                data.See scikit learn
+                Otherwise, this determines the type of averaging performed on
+                the data.See scikit learn
```
Comment (rfratila) [inline vulcanai/models/metrics.py]: should we provide a link or nah?

File: vulcanai/models/metrics.py [hunk 9]
```diff
@@ -615,14 +620,16 @@ def _run_test_multi(network, data_loader, plot=False, save_path=None,
...
+        num_classes = network.num_classes
 
         if num_classes > 2:
             average = "macro"
         elif num_classes == 2:
             logger.warning("Will report scores only for pos_label, which is \
                            set to {}".format(pos_label))
             average = "binary"
+        else:
+            raise ValueError("Incorrect number of classes for run_test_multi")
```
Comment (rfratila) [inline vulcanai/models/metrics.py]: I think you should print what was received and what should be expected to make this error more actionable - or at least use that info to further debug

File: vulcanai/models/utils.py [hunk 10]
```diff
@@ -1,65 +1,79 @@
...
+        index_to_iter : string
+            feature to iterate through adjusting values
+        ls_feat_vals : list
+            values to iterate through for feature in index_to_iter
+
     Returns:
-	dct_scores : dictionary
-	    dictionary of scores
-    """
+    dct_scores : dictionary
```
Comment (rfratila) [inline vulcanai/models/utils.py]: this should be indented over

File: vulcanai/plotters/__init__.py [hunk 11]
```diff
@@ -12,6 +8,8 @@
     display_record,
     display_receptive_fields
 )
+import matplotlib
+matplotlib.use('agg')
```
Comment (rfratila) [inline vulcanai/plotters/__init__.py]: we want to make sure the `plt` import happens after the matplotlib backend get's set otherwise it won't change it after `plt` has been imported.

File: vulcanai/models/basenetwork.py (hunk 2 above)
Comment (nirtiac) [reply vulcanai/models/basenetwork.py]: yes, and not direclty used in this class.

File: vulcanai/models/basenetwork.py (hunk 5 above)
Comment (nirtiac) [reply vulcanai/models/basenetwork.py]: no yet, but it's not used in this function as Joe implemented it, so I wanted to take it out to avoid thoughts that it was used. I want to get rid of it anyways, in a future pull request.

File: vulcanai/models/layers.py (hunk 6 above)
Comment (nirtiac) [reply vulcanai/models/layers.py]: yes.

File: vulcanai/models/metrics.py (hunk 8 above)
Comment (nirtiac) [reply vulcanai/models/metrics.py]: I provided further explanation - it's all documented on the individual metric functions so there's not one link.

File: vulcanai/models/metrics.py (hunk 9 above)
Comment (nirtiac) [reply vulcanai/models/metrics.py]: this would never really happen unless someone overwrote the code. but I added more detail as well.

File: vulcanai/plotters/__init__.py (hunk 11 above)
Comment (nirtiac) [reply vulcanai/plotters/__init__.py]: ok. just reverted .

File: vulcanai/models/layers.py (hunk 7 above)
Comment (nirtiac) [reply vulcanai/models/layers.py]: done
